core/src/main/java/com/kin/ecosystem/core/data/auth/AuthLocalData.java

```diff
@@ -60,6 +60,23 @@ public void setSignInData(@NonNull final SignInData signInData) {
 		editor.apply();
 	}
 
+	@Override
+	public SignInData getSignInData() {
+		final String deviceID = signInSharedPreferences.getString(DEVICE_ID_KEY, null);
+		final String publicAddress = signInSharedPreferences.getString(PUBLIC_ADDRESS_KEY, null);
+		final String signInType = signInSharedPreferences.getString(TYPE_KEY, null);
+		final String jwt = signInSharedPreferences.getString(JWT_KEY, null);
+		final String userID = signInSharedPreferences.getString(USER_ID_KEY, null);
+		final String appID = signInSharedPreferences.getString(APP_ID_KEY, null);
+		if (deviceID != null && publicAddress != null && signInType != null && jwt != null && userID != null
+			&& appID != null) {
+			return new SignInData().deviceId(deviceID).walletAddress(publicAddress)
```

Comment:
you can serialize and descrilze SignInData to/from json string and get/set signInSharedPreferences once

Reply:
?